extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -287,40 +317,49 @@ private boolean isPositionalChannelLinked() {
     public void triggerEvent(String channelId, String event) {
         if (getThing().getChannel(channelId) != null) {
             triggerChannel(getThing().getChannel(channelId).getUID(), event);
-        } else {
-            logger.warn("Event {} in thing {} does not exist, please recreate the thing", event, getThing().getUID());
+            return;
         }
+        logger.warn("Event {} in thing {} does not exist, please recreate the thing", event, getThing().getUID());
     }
 
     /**
-     * Returns the scheduler for the astro jobs.
+     * Returns the scheduler for the Astro jobs wrapped in {@link Optional} instance
      */
-    public Scheduler getScheduler() {
-        return quartzScheduler;
+    public ExpressionThreadPoolExecutor getScheduler() {
+        return scheduledExecutor;
+    }
+    
+    /**
+     * Adds the provided {@link Job} to the Queue (cannot be {@code null})
+     */
+    public void addJobToQueue(Job job) {
```

Comment:
See my comment above, this method should return a boolean stating if the job was added or not.